src/features/common/Modal.test.tsx

```diff
@@ -0,0 +1,37 @@
+import Modal from "./Modal";
+import { expect, describe, test, vi, beforeEach } from "vitest";
+import { render, screen, cleanup, fireEvent } from "~/utils/test/test-utils";
+
+describe("Modal component", () => {
+  beforeEach(() => {
+    cleanup();
+  });
+  test("should render modal when open & not when closed", () => {
+    render(<Modal open={true} data-testid="modal" />);
+    expect(screen.getByTestId("modal")).toBeDefined();
+    cleanup();
+    render(<Modal open={false} data-testid="modal" />);
+    const modal = screen.queryByTestId("modal");
+    expect(modal).not.toBeInTheDocument();

```

Comment:
creo se podria usar el mismo `modal ` para los 2 expects
```suggestion
    const modal = screen.queryByTestId("modal");
    expect(modal).toBeDefined();
    cleanup();
    render(<Modal open={false} data-testid="modal" />);
    expect(modal).not.toBeInTheDocument();
```

Reply:
Pero ahi no hay un re-render en el medio? o sea.. son 2 distintos creo. Igual si quedo raro que uno esta como const y otro dentro de la funcion

Reply:
me quedaria la duda de si `modal` queda haciendo referencia al primero que se renderiza, o si se vuelve a definir cada vez que se usa? porque lo probe y el test no fallaba

Reply:
Ah si no falla entonces mejor, si